octobot_tentacles_manager/managers/python_modules_requirements_manager.py

```diff
@@ -0,0 +1,136 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+import os
+import sys
+
+from octobot_commons.enums import OctoBotTypes
+from octobot_commons.logging.logging_util import get_logger
+from octobot_commons.os_util import get_octobot_type
+
+
+# TODO: adapt this wrapper to handle python modules requirements
+# Warning: Impossible to dynamically load python modules with unmet dependencies in frozen python environment
+# (ex: compiled binary). However these dynamic imports work with Python / Docker environment.
+# code from pip POC (https://github.com/Drakkar-Software/OctoBot-Tentacles-Manager/tree/pip-usage-poc)
+
+# TODO: remove from to .coveragerc when adapted
+
+"""
+Tentacles management
+"""
+
+
+async def install_tentacle(tentacle_path: str, tentacle_name: str) -> (str, str):
+    return await _run_pip_install(os.path.join(tentacle_path, "Trading"), tentacle_name)
+
+
+async def update_tentacle(tentacle_path: str, tentacle_name: str) -> (str, str):
+    return await _run_pip_update(os.path.join(tentacle_path, "Trading"), tentacle_name)
+
+
+async def list_installed_tentacles(tentacle_path: str) -> list:
+    return [await _run_pip_freeze(os.path.join(tentacle_path, tentacle_type_path))
+            for tentacle_type_path in ["Trading"]]
+
+"""
+Pip wrapper
+"""
+
+
+async def _run_pip_command(args) -> (str, str):
+    if get_octobot_type() == OctoBotTypes.BINARY.value:
+        raise RuntimeError("Can't use PIP in a frozen binary environment")
```

Comment:
It will stop the whole process, do we really want this to happend ?

Reply:
I think it should be up to the caller to chose either to catch this exception or not, for now it's never called